genesis/vis/rasterizer_context.py

```diff
@@ -412,9 +412,11 @@ def on_mpm(self):
 
     def update_mpm(self, buffer_updates):
         if self.sim.mpm_solver.is_active():
-            particles_all = self.sim.mpm_solver.particles_render.pos.to_numpy()
-            active_all = self.sim.mpm_solver.particles_render.active.to_numpy().astype(bool)
-            vverts_all = self.sim.mpm_solver.vverts_render.pos.to_numpy()
+            particles_all = self.sim.mpm_solver.particles_render.pos.to_numpy()[:, self.rendered_envs_idx[0]]
```

Comment:
create local variable `particles_render = self.sim.mpm_solver.particles_render` on top and use it everywhere. This is just unreadable.